lib/loc.ml

```diff
@@ -13,6 +13,7 @@
 type source =
   | InFile of string
   | ToplevelInput
+  | Initialization
 
```

Comment:
I am positive this is not the way to go, IMHO locations should refer to file locations, what does the fact that system initialization has to do with where an object is located?

Reply:
I agree that a phase of an execution is not of the same nature as a location in a text. Locations are used in the step interpretation a sentence and are not used in the initialization step. Would a type of the form:
```ocaml
type initialization_step = Configuring | RequiringFile of dirpath | ...
type phase = Initialization of initialization_step | SentenceInterpretation of Location.t
```
be better for you?

Then, the warning printer and the main error printer would use this information to `locate_if_not_already` and/or add an appropriate phase message.

Or do you have something else in mind?

Reply:
IMHO the lower layers should know nothing about the phase, letting them know is just leakage, It is the top-level handler that has to be aware, so I don't see the point of introducing all such types. IMHO this PR is getting a bit complicated for what it should a 20 lines patch at most.

Reply:
My intent with this PR was also to address the question of warnings.

For errors, I could move the information of the phase in `stm.ml` (would it be ok at the level of `process_transaction`?).

For warnings, I have to propagate the information one way or another (even if in another PR), and I don't see how to do it differently than by using a type such as `phase`...

Reply:
I see, I had looked at the original PR description which talks only about errors. So for the warnings, a few options are possible, but indeed this could be a good opportunity to get rid of the hideous set location hack and instead let `coqloop_feed` handle it.

I think the warnings change is fully orthogonal and other developers may want to get involved so I suggest we do it in a different PR and we get the fix for errors merged here.